cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/ApplicationVerificationStage.java

```diff
@@ -122,8 +126,19 @@ public void process(ApplicationDeployable input) throws Exception {
     emit(input);
   }
 
-  private void verifySpec(ApplicationId appId,
-                          ApplicationSpecification specification) {
+  private void compareSpecWithLatest(ApplicationId appId, ApplicationSpecification specification) {
+    ApplicationMeta latestApp = store.getLatestAppVersion(appId);
+    if (latestApp == null || latestApp.getSpec() == null)  {
+      return;
+    }
+
+    if (specification.getConfiguration().equals(latestApp.getSpec().getConfiguration())) {
```

Comment:
Should not compare the configuration since it is a json string so will be super volatile, better to compare the actual spec inside the spec, you can create a JIRA and do it in later pr